lib/class-wp-theme-json-gutenberg.php

```diff
@@ -18,6 +18,14 @@ class WP_Theme_JSON_Gutenberg {
 	 */
 	private $theme_json = null;
 
+	/**
+	 * What source of data this object represents.
+	 * One of VALID_ORIGINS.
+	 *
+	 * @var string
+	 */
+	private $origin = null;
```

Comment:
What happens if this class contains the result of theme.json from multiple origins e.g: if it is the result of core merged with theme and user? In that case (which is common) the value stored as the origin is not something useful as there is no single origin. Given that in an object we may have multiple origins I think the class should not contain an origin field and a get origin method.

Reply:
This was bothering me as well. Luckily, we don't need it. See https://github.com/WordPress/gutenberg/pull/32622#discussion_r650084806